app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/stats/VpnServiceStateLogger.kt

```diff
@@ -16,49 +16,95 @@
 
 package com.duckduckgo.mobile.android.vpn.stats
 
+import android.annotation.SuppressLint
+import com.duckduckgo.app.utils.ConflatedJob
+import com.duckduckgo.appbuildconfig.api.AppBuildConfig
 import com.duckduckgo.di.scopes.VpnScope
+import com.duckduckgo.mobile.android.vpn.model.AlwaysOnState
 import com.duckduckgo.mobile.android.vpn.model.VpnServiceState
 import com.duckduckgo.mobile.android.vpn.model.VpnServiceStateStats
 import com.duckduckgo.mobile.android.vpn.model.VpnStoppingReason
 import com.duckduckgo.mobile.android.vpn.model.VpnStoppingReason.ERROR
 import com.duckduckgo.mobile.android.vpn.model.VpnStoppingReason.REVOKED
 import com.duckduckgo.mobile.android.vpn.model.VpnStoppingReason.SELF_STOP
 import com.duckduckgo.mobile.android.vpn.model.VpnStoppingReason.UNKNOWN
+import com.duckduckgo.mobile.android.vpn.service.TrackerBlockingVpnService
 import com.duckduckgo.mobile.android.vpn.service.VpnServiceCallbacks
 import com.duckduckgo.mobile.android.vpn.state.VpnStateMonitor.VpnStopReason
 import com.duckduckgo.mobile.android.vpn.store.VpnDatabase
 import com.squareup.anvil.annotations.ContributesMultibinding
 import dagger.SingleInstanceIn
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.delay
 import kotlinx.coroutines.launch
 import timber.log.Timber
 import java.util.concurrent.Executors
 import javax.inject.Inject
+import javax.inject.Provider
 
 @ContributesMultibinding(
     scope = VpnScope::class,
     boundType = VpnServiceCallbacks::class
 )
 @SingleInstanceIn(VpnScope::class)
-class VpnServiceStateLogger @Inject constructor(private val vpnDatabase: VpnDatabase) : VpnServiceCallbacks {
+class VpnServiceStateLogger @Inject constructor(
+    private val vpnDatabase: VpnDatabase,
+    private val vpnService: Provider<TrackerBlockingVpnService>,
+    private val appBuildConfig: AppBuildConfig,
+) : VpnServiceCallbacks {
 
     private val dispatcher = Executors.newSingleThreadExecutor().asCoroutineDispatcher()
+    private val job = ConflatedJob()
 
     override fun onVpnStarted(coroutineScope: CoroutineScope) {
-        coroutineScope.launch(dispatcher) {
+        job += coroutineScope.launch(dispatcher) {
             Timber.d("VpnServiceStateLogger, new state ENABLED")
             vpnDatabase.vpnServiceStateDao().insert(VpnServiceStateStats(state = VpnServiceState.ENABLED))
+
+            @SuppressLint("NewApi") // IDE doesn't get we use appBuildConfig
+            if (appBuildConfig.sdkInt >= 29) {
+                incrementalPeriodicChecks {
```

Comment:
The reason for this is, we need to check periodically whether always-on lockdown mode is enabled. But it's hard to pick one period because:
* we want something that is frequent enough so that if user enables AppTP -> then sees the always on promotion -> enables always-on and lock down, we are able to detect it swiftly
* we want something that is not so frequent so that we don't do extra IO unnecessary IO operations

The solution is this `incrementalPeriodicChecks` that starts with an initial period of 0.5s, increments by 10% for every check and the max period is 5min